client/rest-high-level/src/test/java/org/elasticsearch/client/core/BroadcastResponseTests.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.core;
+
+import org.elasticsearch.action.support.DefaultShardOperationFailedException;
+import org.elasticsearch.cluster.ClusterModule;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContent;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.seqno.RetentionLeaseNotFoundException;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.isIn;
+
+public class BroadcastResponseTests extends ESTestCase {
+
+    public void testFromXContent() throws IOException {
+        final String index = randomAlphaOfLength(8);
```

Comment:
Maybe use `AbstractXContentTestCase#xContentTester()` static method to test xcontent serialization?
I think it is a bit shorter and we have more test iterations by default.

Reply:
I think that `AbstractXContentTestCase` here is a lie. This is because what we really want to test is that if the REST layer sends back an `XContent` serialized `org.elasticsearch.action.support.broadcast.BroadcastResponse` then `org.elasticsearch.client.core.BroadcastResponse` is able to de-serialize this `XContent`. `AbstractXContentTestCase` does not allow to test such a situation, instead only allowing you to go from concrete instance (of `org.elasticsearch.client.core.BroadcastResponse`) to `XContent` and then back to a concrete instance of (`org.elasticsearch.client.core.BroadcastResponse`), and then checks if the result is equal. That is not what actually happens in production though, what happens in production is what I outlined above: a concrete instance of `org.elasticsearch.action.support.broadcast.BroadcastResponse` is serialized on the REST layer which then has to be de-serialized on the HL client layer by `org.elasticsearch.client.core.BroadcastResponse`. Hence, this test reflects that.

Reply:
Good point. I think we should open an issue, so that goes over the existing hlrc response serialization tests and change those tests to test way is tested here.